datadog/resource_datadog_integration_webhook_test.go

```diff
@@ -0,0 +1,600 @@
+			expectedTerraformHeaders: map[string]interface{}{},
+			expectedError:            nil,
+		},
+		"no headers with whitespace": {
+			datadogHeaders:           "  \t",
+			expectedTerraformHeaders: map[string]interface{}{},
+			expectedError:            nil,
+		},
+		"single header": {
+			datadogHeaders: "header1: val1",
+			expectedTerraformHeaders: map[string]interface{}{
+				"header1": "val1",
+			},
+			expectedError: nil,
+		},
+		"multiple headers": {
+			datadogHeaders: "header1: val1\nheader2: val2\nheader3: val3",
+			expectedTerraformHeaders: map[string]interface{}{
+				"header1": "val1",
+				"header2": "val2",
+				"header3": "val3",
+			},
+			expectedError: nil,
+		},
+		"colon in value": {
+			datadogHeaders: "header1: val:with:colon",
+			expectedTerraformHeaders: map[string]interface{}{
+				"header1": "val:with:colon",
+			},
+			expectedError: nil,
+		},
+		"no colon between header and value": {
+			datadogHeaders:           "header1 val",
+			expectedTerraformHeaders: nil,
+			expectedError:            fmt.Errorf("header not correctly formatted, expected ':' in 'header1 val'"),
+		},
+	}
+	for name, tc := range cases {
+		actualTerraformHeaders, err := buildTerraformHeader(tc.datadogHeaders)
+
+		if err != nil && tc.expectedError == nil {
+			t.Errorf("%s: Unexpected error occured '%s", name, err.Error())
+		} else if err == nil && tc.expectedError != nil {
+			t.Errorf("%s: Expected error '%s', but it was not thrown", name, tc.expectedError.Error())
+		} else if err != nil && tc.expectedError != nil && err.Error() != tc.expectedError.Error() {
+			t.Errorf("%s: Unexpected error occured '%s', expected '%s'", name, err.Error(), tc.expectedError.Error())
+		} else if actualTerraformHeaders == nil && tc.expectedError == nil {
+			t.Errorf("%s: Did not expect an error and terraform webhook is nil", name)
+		} else if actualTerraformHeaders != nil && tc.expectedError != nil {
+			t.Errorf("%s: Expected error '%s', but got non-nil terraform webhook", name, tc.expectedError.Error())
+		} else if actualTerraformHeaders != nil {
+			compareTerraformHeaders(tc.expectedTerraformHeaders, *actualTerraformHeaders, name, t)
+		}
+	}
+}
+
+func TestBuildDatadogWebhook(t *testing.T) {
+	cases := map[string]struct {
+		terraformWebhook       map[string]interface{}
+		expectedDatadogWebhook datadog.Webhook
+	}{
+		"required fields only": {
+			terraformWebhook: map[string]interface{}{
+				"name": "my_webhook",
+				"url":  "http://example.com",
+			},
+			expectedDatadogWebhook: datadog.Webhook{
+				Name: datadog.String("my_webhook"),
+				URL:  datadog.String("http://example.com"),
+			},
+		},
+		"custom headers": {
+			terraformWebhook: map[string]interface{}{
+				"name": "my_webhook",
+				"url":  "http://example.com",
+				"headers": map[string]interface{}{
+					"header1": "val1",
+				},
+			},
+			expectedDatadogWebhook: datadog.Webhook{
+				Name: datadog.String("my_webhook"),
+				URL:  datadog.String("http://example.com"),
```

Comment:
🤔 why are not `Headers: datadog.String("header1: val),` expected?


```suggestion
				URL:  datadog.String("http://example.com"),
				Headers: datadog.String("header1: val),
```

Reply:
yep that was a miss, good catch